### PR TITLE
Add raycasting against tools

### PR DESCRIPTION
**Pull request by dangond-ptc**
Requires https://github.com/ptcrealitylab/vuforia-spatial-edge-server/pull/1091

### CODE REVIEW

File: src/device/index.js [hunk 1]
```diff
@@ -392,6 +406,74 @@ realityEditor.device.postEventIntoIframe = function(event, frameKey, nodeKey) {
...
+            let resolved = false;
+
+            const key = object + frame + currentIndex;
+            setTimeout(() => {
+                realityEditor.device.raycastCallbacks.delete(key);
+                if (!resolved) {
+                    resolved = true;
+                    reject();
+                }
+            }, 200); // Reject automatically within 200 ms
```
Comment (Steve-KX-RL) [inline src/device/index.js:445]: https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/assets/111802681/3759d415-6593-47de-a46f-07b8907dfc66

Looks like when I draw very fast, it raycasts on the table first and then flies off. When I draw slower, it forms these squiggly lines. Is it due to this 200ms interval where we decide to listen for tool raycasts?

Comment (dangond-ptc) [reply src/device/index.js:445]: Close, that's a side-effect of how the drawing tool draws that I overlooked when adding the caching mechanism on the object.js side in the edge-server. Removing the caching doesn't seem to cause any performance issues, so I just removed that instead. This is just here to ensure a badly-behaved tool doesn't cause tons of touch events to be delayed.

Comment (Steve-KX-RL) [reply src/device/index.js:445]: Got it. I pulled that commit and no longer get this artifact.

**Review by Steve-KX-RL — APPROVED** [review]
LGTM!

File: src/device/index.js [hunk 2]
```diff
@@ -392,6 +406,74 @@ realityEditor.device.postEventIntoIframe = function(event, frameKey, nodeKey) {
...
+
+            realityEditor.device.raycastCallbacks.set(key, (result) => {
+                if (!resolved) {
+                    resolved = true;
+                    resolve(result);
+                }
+            });
+        })
+    });
+    const raycastResults = await Promise.all(frameRaycastPromises);
```
Comment (Steve-KX-RL) [inline src/device/index.js:455]: Just leave it here for potential consideration: this line alone spends ~2-4 ms of our performance budget (~16.7 ms in total if we're to maintain 60 fps). So we could potentially have a flag in device/index.js that gets set when one tool has enabled drawing / raycasting inside of it, and if no tool is enabled, we can skip this entire section.

Comment (ddgond) [reply src/device/index.js:455]: It's async, so unless I'm mistaken, this is just a 2-4ms latency between a mouse click and the tool receiving the event info. It should not be blocking the event loop/taking up our performance budget.

Comment (Steve-KX-RL) [reply src/device/index.js:455]: Oh make sense

**Pull request merged**